CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,399 @@
+
+# Future
+
+After opening a pull request it will be reviewed and you will likely be asked to make some changes. This is normal and also happens with pull requests submitted by experienced contributors.
+
+Changes typically include improving code style, tweaking phrasing and quest settings.
+
+You are also welcome to help with reviewing other PRs - different people have different strengths, there are active reviewers who help with code style, there are some native speakers of English, some with deep knowledge of OSM tagging schemes or deep knowledge of how StreetComplete works.
+
+After the PR is finished it will be merged before the beta release of the next version. This way it can be additionally tested with a wider audience before release to all and translators can [translate text into other languages](CONTRIBUTING.md#translating-the-app).
+
+After full release it will reach the entire StreetComplete audience who now will be able to more easily contribute to OpenStreetMap. Thanks in advance for that!
+
+# Bad documentation is a bug
+
+Unclear documentation, including this one, is a bug. Feel free to either submit a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) with a fix or [open an issue](https://github.com/streetcomplete/StreetComplete/issues/new) describing your confusion.
+
+Note that not everything will be directly described. This document is very intentionally not including step-by-step guide to install Android Studio, [linking](CONTRIBUTING.md#development) to official docs instead.
+
+# More complexity
+What was described above is an attempt to cover all aspects of quest creation, without describing all the complexity.
+
+Below is some additional info.
+
+## getTitle Parameters
+In some cases, the quest will mention the name and type of feature. For example, in the case of shops to make their identification possible.
+
+This requires preparing space in the message for filling at runtime,
+
+And to add the mechanism supplying this data. Here is [some typical code](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/wheelchair_access/AddWheelchairAccessBusiness.kt#L102-L111) that will:
+
+```
+    override fun getTitle(tags: Map<String, String>) =
+        if (hasFeatureName(tags))
+            R.string.quest_wheelchairAccess_name_type_title
+        else
+            R.string.quest_wheelchairAccess_name_title
+
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        val name = tags["name"] ?: tags["brand"]
+        return arrayOfNotNull(name, featureName.value)
+    }
+``` 
+
+- try to pass name, feature type (such as "greengrocer", translated to the proper language)
```

Comment:
```suggestion
- try to pass name or feature type (such as "greengrocer", translated to the proper language)
```
```suggestion
- try to pass name and feature type (such as "greengrocer", translated to the proper language)
```

Reply:
rewriting it substantially, thanks for pointing to a confused section